Sources/Testing/Running/Configuration.swift

```diff
@@ -128,17 +128,38 @@ public struct Configuration: Sendable {
   ///
   /// By default, all tests are run and no filter is set.
   public mutating func setTestFilter(toMatch selection: Set<Test.ID>?) {
-    setTestFilter(toMatch: selection.map(Test.ID.Selection.init))
+    setTestFilter(toMatch: selection, includeHiddenTests: false)
   }
-  
+
+  /// The granularity to enforce test filtering.
+  /// - Parameters:
+  ///   - selection: A set of test IDs to be filtered. If `nil`, the current
+  ///     selection is cleared.
+  ///   - includeHiddenTests: If false, a test annotated with the `.hidden` trait will not be included, even if its ID is present in `selection`.
+  ///
+  /// By default, all tests are run and no filter is set.
+  mutating func setTestFilter(toMatch selection: Set<Test.ID>?, includeHiddenTests: Bool) {
+    setTestFilter(toMatch: selection.map(Test.ID.Selection.init), includeHiddenTests: includeHiddenTests)
+  }
+
   /// The granularity to enforce test filtering.
   ///
   /// - Parameters:
   ///   - selection: A selection of test IDs to be filtered. If `nil`, the
   ///     current selection is cleared.
+  ///   - includeHiddenTests: If false, a test annotated with the `.hidden` trait will not be included, even if its ID is present in `selection`.
   ///
   /// By default, all tests are run and no filter is set.
-  mutating func setTestFilter(toMatch selection: Test.ID.Selection?) {
-    testFilter = selection.map { $0.contains }
+  mutating func setTestFilter(toMatch selection: Test.ID.Selection?, includeHiddenTests: Bool) {
```

Comment:
Do we need _both_ new overloads for this internal code? Are we using both?

Reply:
We don't. I'll eliminate the redundancy.